lib/ruby_lsp/response_builder.rb

```diff
@@ -0,0 +1,50 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  class OperationNotPermitted < StandardError; end
+
+  class ResponseBuilder
```

Comment:
Yeah, I agree we should unify the abstract classes. 

I'll make the change, but to your point, I definitely think we should narrow in to a common builder class before we proceed with the release.

Reply:
I decided to adopt the builder concept in #1325 (with a slightly different form of structure). 

But I still think we should not add more common APIs before we've converted most requests. For example, we probably don't want addons to have the `pop` API for semantic highlighting.

Reply:
I think it would be best to finalize on an approach for the builder in this PR, get this merged, then you can rebase in your PR. I can factor in your approach and modify mine accordingly. 

Thoughts? Just thinking of the best way to minimize conflicts.